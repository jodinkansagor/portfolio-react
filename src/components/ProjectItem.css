.projectItem {
  margin-top: 15px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;


  img {
    width: 80%
  }

  .links {
    margin-top: 7px;
    color: #d9b08c;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-end;
    
    p {
      width: 12px;
      margin-left: 5px;
      margin-right: 5px;
    }

    a {
      color: #d9b08c
    }

    a:hover {
      color: #116466
    }
  }

  p {
    
    margin: 0px;
    margin-top: 7px;
    color: #2c3531;
    width: 80%
  }

  .stack {
    color: #116466
  }
}

