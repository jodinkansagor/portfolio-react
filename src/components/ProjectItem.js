import React from 'react';
import PropTypes from 'prop-types';
import styles from '../components/ProjectItem.css';


const ProjectItem = ({ img, description, stack, siteLink, githubLink }) => {
  return (
    <section className={styles.projectItem}>
      <img src={img} />
      <p>{description}</p>
      <p className={styles.stack}>{stack}</p>
      <section className={styles.links}>
        <a href={siteLink} target="_blank">site{' '}</a>
        <p>//</p>
        <a href={githubLink} target="_blank">github</a>
      </section>

    </section>
  );
};

ProjectItem.propTypes = {
  img: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  stack: PropTypes.string.isRequired,
  siteLink: PropTypes.string.isRequired,
  githubLink: PropTypes.string.isRequired
}

export default ProjectItem;
