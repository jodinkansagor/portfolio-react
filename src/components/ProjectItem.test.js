import React from 'react';
import { shallow } from 'enzyme';
import ProjectItem from './ProjectItem';

describe('ProjectItem component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<ProjectItem img={'img'} description={'description'} stack={'stack'} siteLink={'sitelink'} githubLink={'githublink'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
