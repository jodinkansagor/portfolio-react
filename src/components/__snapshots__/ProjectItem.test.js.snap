// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem component matches a snapshot 1`] = `
<section
  className="projectItem"
>
  <img
    src="img"
  />
  <p>
    description
  </p>
  <p
    className="stack"
  >
    stack
  </p>
  <section
    className="links"
  >
    <a
      href="sitelink"
      target="_blank"
    >
      site
       
    </a>
    <p>
      //
    </p>
    <a
      href="githublink"
      target="_blank"
    >
      github
    </a>
  </section>
</section>
`;
